### PR TITLE
Vetal

### PR DESCRIPTION
**Pull request by Greifelt228**

### CODE REVIEW

**Comment by Iangyl** [comment]
make self-describing commits and fix deploy errors, please

File: common/components/CommentsCard/index.tsx [hunk 1]
```diff
@@ -44,24 +44,30 @@ const CommentsCard: React.FC<Props> = ({ taskId, loading = false }) => {
...
+    <div className={s.CardDiv}>
+      <Card loading={loading} className={s.Card} title="Коментарі">
+        <List
+          className={s.List}
+          dataSource={task?.data?.comment}
+          renderItem={(item, index) => (
+            <List.Item key={index}>
+              <Comment comment={item} taskId={task?.data?._id} />
+              <div ref={bottomRef} />
+            </List.Item>
```
Comment (Iangyl) [inline common/components/CommentsCard/index.tsx:67]: This code fragment:
```
<List.Item key={index}>
    <Comment comment={item} taskId={task?.data?._id} />
    <div ref={bottomRef} />
</List.Item>
```
is equal to the next one:
```
<List.Item key={index}>
    <Comment ref={bottomRef} comment={item} taskId={task?.data?._id} />
</List.Item>
```
or
```
<List.Item key={index} ref={bottomRef} >
    <Comment comment={item} taskId={task?.data?._id} />
</List.Item>
```
so this `<div>` is unnecessary.

File: common/components/DashboardPage/blocks/categories.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const router = useRouter()
+  const tasksResponse = useGetAllTaskQuery('')
+  const tasks = tasksResponse?.data?.data
+  const { data: categoriesData } = useGetAllCategoriesQuery('')
+  const categories = categoriesData?.data
+
+  const getCount = (name) => {
+    return tasks?.filter((task) => task?.category == name)
+  }
+  // console.log(getCount('Vodniki1').length)
```
Comment (Iangyl) [inline common/components/DashboardPage/blocks/categories.tsx]: remove pls

File: common/components/DashboardPage/blocks/categories.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const session = useSession()
+  const router = useRouter()
+  const tasksResponse = useGetAllTaskQuery('')
+  const tasks = tasksResponse?.data?.data
+  const { data: categoriesData } = useGetAllCategoriesQuery('')
+  const categories = categoriesData?.data
+
+  const getCount = (name) => {
+    return tasks?.filter((task) => task?.category == name)
+  }
```
Comment (Iangyl) [inline common/components/DashboardPage/blocks/categories.tsx:25]: memoize it

File: common/components/DashboardPage/blocks/categories.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    },
+    {
+      title: 'Кількість',
+      dataIndex: 'taskincategory',
+      key: 'taskincategory',
+      width: '30%',
+      // render: (taskincategory) =>
+      //   taskincategory !== [] ? taskincategory.length : getCount(name).length,
+    },
+  ]
```
Comment (Iangyl) [inline common/components/DashboardPage/blocks/categories.tsx:50]: eject it to separate file, please

File: pages/category/index.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  const categories = categoriesData?.data
+
+  return (
+    <>
+      <List
+        header={<div>Всі категорії</div>}
+        bordered
+        dataSource={categories}
+        renderItem={(item) => (
+          <List.Item>
```
Comment (Iangyl) [inline pages/category/index.tsx:26]: add `key` value here

File: pages/category/[id].tsx [hunk 6]
```diff
@@ -29,7 +29,7 @@ const Category: React.FC = () => {
   const dataSource = useMemo(() => {
     return tasks?.filter((task) => task?.category === categoryTask?.name)
   }, [tasks, categoryTask?.name])
-  console.log('dataS', dataSource)
+  // console.log('dataS', dataSource)
```
Comment (Iangyl) [inline pages/category/[id].tsx]: remove it, please

File: common/components/DashboardPage/blocks/categories.tsx [hunk 7]
```diff
@@ -22,22 +23,29 @@ const CategoriesBlock: React.FC<{ style: string }> = ({ style }) => {
   const getCount = (name) => {
     return tasks?.filter((task) => task?.category == name)
   }
-  // console.log(getCount('Vodniki1').length)
+  // console.log(getCount('Vodniki1'))
```
Comment (Iangyl) [inline common/components/DashboardPage/blocks/categories.tsx:26]: the same here

File: common/components/DashboardPage/blocks/categories.tsx [hunk 8]
```diff
@@ -22,22 +23,29 @@ const CategoriesBlock: React.FC<{ style: string }> = ({ style }) => {
...
-      key: 'taskincategory',
-      width: '30%',
-      // render: (taskincategory) =>
-      //   taskincategory !== [] ? taskincategory.length : getCount(name).length,
+      dataIndex: 'name',
+      key: 'name',
+      width: '20%',
+      render: (name) => getCount(name)?.length,
     },
   ]
```
Comment (Iangyl) [inline common/components/DashboardPage/blocks/categories.tsx:50]: eject it to separate file, please

**Review by Iangyl — CHANGES_REQUESTED** [review]

File: utils/helpers.ts [hunk 9]
```diff
@@ -1,2 +1,6 @@
 export const firstTextToUpperCase = (text: string) =>
   text[0].toUpperCase() + text.slice(1)
+
+export const getCount = (tasks: any, name: string) => {
+  return tasks?.filter((task) => task?.category == name)
+}
```
Comment (Iangyl) [inline utils/helpers.ts:6]: Does it universal function? You should place only universal functions in helpers file or functions which are used in several different places.

File: common/components/DashboardPage/blocks/categories.tsx [hunk 10]
```diff
@@ -20,11 +21,6 @@ const CategoriesBlock: React.FC<{ style: string }> = ({ style }) => {
   const { data: categoriesData } = useGetAllCategoriesQuery('')
   const categories = categoriesData?.data
 
-  const getCount = (name) => {
-    return tasks?.filter((task) => task?.category == name)
-  }
-  // console.log(getCount('Vodniki1'))
-
   const columns = [
```
Comment (Iangyl) [inline common/components/DashboardPage/blocks/categories.tsx:24]: You should eject constants to another file the in next cases:
- const initialization takes up a lot of space
- you have many constants in the file

If this constant don't use anywhere in the code, then create a new file (`categories.config.ts`) near the current and move the constant to it.

File: common/components/TaskCard/index.tsx [hunk 11]
```diff
@@ -36,7 +36,8 @@ const TaskCard = ({ taskId, task }) => {
 
   const taskDelete = (id) => {
     deleteTask(id)
-    router.push(AppRoutes.TASK)
+    // router.push(AppRoutes.TASK)
+    router.push(`/task/user/${user?._id}`)
```
Comment (Iangyl) [inline common/components/TaskCard/index.tsx:40]: change it to ```router.push(`${AppRoutes.TASK}/user/${user?._id}`)``` or create a new `enum` with different URLs and use it, or extend the existed `AppRoutes` `enum`

**Review by Iangyl — CHANGES_REQUESTED** [review]

**Comment by Iangyl** [comment]
and resolve conflicts, please

File: utils/helpers.ts (hunk 9 above)
Comment (Greifelt228) [reply utils/helpers.ts:6]: it is used twice

File: pages/task/index.tsx [hunk 12]
```diff
@@ -11,6 +11,7 @@ import { GetServerSideProps } from 'next'
 import CardOneTask from '../../common/components/CardOneTask'
 import s from './style.module.scss'
 import Filter from '../../common/components/UI/Filtration'
+import { _id } from '@next-auth/mongodb-adapter'
```
Comment (VanyaRPP) [inline pages/task/index.tsx:14]: Remove if u don't use it.

**Pull request merged**